src/scripts/angry-tock.test.js

```diff
@@ -0,0 +1,319 @@
+        username: "employee2",
+      },
+      {
+        id: "tock4",
+        email: "user@four",
+        first_name: "User",
+        last_name: "Four",
+        username: "employee4",
+      },
+    ]);
+
+    get18FTockSlackUsers.mockResolvedValue([
+      {
+        email: "user@one",
+        id: "tock1",
+        name: "User 1",
+        slack_id: "slack1",
+        user: "employee1",
+      },
+      {
+        email: "user@two",
+        id: "tock2",
+        name: "User 2",
+        slack_id: "slack2",
+        user: "employee2",
+      },
+      {
+        email: "user@three",
+        id: "tock3",
+        name: "User 3",
+        slack_id: "slack3",
+        user: "employee3",
+      },
+    ]);
+  });
+
+  afterAll(() => {
+    clock.restore();
+  });
+
+  it("issues a warning and does not schedule a shouting match if un/mis-configured", async () => {
+    process.env.TOCK_API = "";
+    process.env.TOCK_TOKEN = "";
+    let angryTock = await load();
+    angryTock(app);
+    expect(app.logger.warn).toHaveBeenCalled();
+    expect(scheduleJob).not.toHaveBeenCalled();
+
+    process.env.TOCK_API = "set";
+    process.env.TOCK_TOKEN = "";
+    angryTock = await load();
+    angryTock(app);
+    expect(app.logger.warn).toHaveBeenCalled();
+    expect(scheduleJob).not.toHaveBeenCalled();
+
+    process.env.TOCK_API = "";
+    process.env.TOCK_TOKEN = "set";
+    angryTock = await load();
+    angryTock(app);
+    expect(app.logger.warn).toHaveBeenCalled();
+    expect(scheduleJob).not.toHaveBeenCalled();
+  });
+
+  describe("schedules the next shouting match on startup", () => {
+    it("if it is shouty day and before first-shout time, schedules a first-shout", async () => {
+      // Monday, April 8, 1974: Hank Aaron hits his 715th career homerun,
+      // breaking Babe Ruth's record.
+      const time = moment.tz(
+        "1974-04-08 00:00:00",
+        process.env.ANGRY_TOCK_TIMEZONE
+      );
+      clock.tick(time.toDate().getTime());
+
+      const angryTock = await load();
+      angryTock(app);
+
+      time.hour(10);
+      expect(scheduleJob).toHaveBeenCalledWith(
+        time.toDate(),
+        expect.any(Function)
+      );
+    });
+
+    it("if it is shouty day, after the first-shout time but before the second-shout time, schedules a second-shout", async () => {
+      // Monday, May 20, 1991: Michael Jordan named NBA MVP.
+      const time = moment.tz(
+        "1991-05-20 11:00:00",
+        process.env.ANGRY_TOCK_TIMEZONE
+      );
+      clock.tick(time.toDate().getTime());
+
+      const angryTock = await load();
+      angryTock(app);
+
+      time.hour(14);
+      time.minute(45);
+      expect(scheduleJob).toHaveBeenCalledWith(
+        time.toDate(),
+        expect.any(Function)
+      );
+    });
+
+    it("if it is shouty day, after the second-shout time, schedules a first-shout for the next shouty day", async () => {
+      // Monday, January 20, 1997 - Bill Clinton is sworn into his second term
```

Comment:
This one is quirky because it's _only_ a holiday for them. Shouty-bot is not location-aware, but that could be a neat enhancement.

Also, I have to admit I only picked this date because I was looking for a significant Monday between 1991 and 2019 (so it fit chronologically between the previous and following tests). I picked 1997 at random and started scanning Wikipedia. My criteria was a Monday that was not a holiday where the next Monday was also not a holiday, and relatively significant events happened on both Mondays. I picked the first one I found. 😂

Reply:
actually this makes it a better long-term test maybe-- if shouty bot ever becomes location-aware then it should fail!